include/effects/matrix/PatternWeather.h

```diff
@@ -350,45 +408,81 @@ class PatternWeather : public LEDStripEffect
             vTaskDelay(pdMS_TO_TICKS(WEATHER_CHECK_WIFI_WAIT));
         }
 
-        updateCoordinates();
+        // Only try to update if we have an API Key
+        if (!g_ptrSystem->DeviceConfig().GetOpenWeatherAPIKey().isEmpty())
+        {
+            updateCoordinates();
 
-        if (getWeatherData())
-            getTomorrowTemps(highTomorrow, loTomorrow);
+            if (getWeatherData())
+                getTomorrowTemps(highTomorrow, loTomorrow);
+        }
     }
 
+    /**
+     * @brief Determine if the device configuration has changed the
+     * location selection
+     * 
+     * @return bool - true if the location has changed 
+     */
     bool HasLocationChanged()
     {
-        String configLocation = g_ptrSystem->DeviceConfig().GetLocation();
-        String configCountryCode =g_ptrSystem->DeviceConfig().GetCountryCode();
+        bool locationChanged = g_ptrSystem->DeviceConfig().GetLocation() != strLocation;
+        bool contryChanged = g_ptrSystem->DeviceConfig().GetCountryCode() != strCountryCode;
 
-        return strLocation != configLocation || strCountryCode != configCountryCode;
+        return locationChanged || contryChanged;

```

Comment:
I missed this earlier today: "contryChanged" should be "co**u**ntryChanged".

Reply:
I don't know how I missed it too.  I guess because it still ran properly and didn't throw an error I just glossed bver it.